cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -24,11 +24,11 @@
 #include <arcticdb/python/gil_lock.hpp>
 
 namespace arcticdb::version_store {
-
+template<class ClockType>
 LocalVersionedEngine::LocalVersionedEngine(
         const std::shared_ptr<storage::Library>& library,
-        const std::optional<std::string>& license_key ARCTICDB_UNUSED) :
-    store_(std::make_shared<async::AsyncStore<util::SysClock>>(library, codec::default_lz4_codec(), encoding_version(library->config()))),
+        const ClockType&) :
```

Comment:
The `ClockType` param isn't necessary

Reply:
It's used to allow the `ClockType` template variable to be inferred, as there's no syntax to do this explicitly.

Reply:
OK. There are plenty of other types that inject a clock without using this pattern, I would prefer that we're consistent with them.

Reply:
This case is unique in that the `ClockType` is only used in the constructor itself and never in the body of the class.

In other words, I only wanted a templated constructor function, not a templated class which requires too many changes. (Every place that used to refer to `LocalVersionedEngine` and its sub-type `PythonVersionStore` will have to become transitively templated.)

The most sensible API is to make the c'tor accept `shared_ptr<Stroages>`, which would allow the Clock to be picked external to the c'tor, but that would still involve a lot of refactoring.

Reply:
We could make it a variant, that's the normal way to avoid transitive type parameterization

Reply:
I'll open a separate PR to fix this